nuxeo-routing-rest-api/src/test/java/org/nuxeo/ecm/restapi/server/WorkflowEndpointTest.java

```diff
@@ -776,6 +778,51 @@ public void testResumeWorkflowWithDeletedAttachedDoc()
 
     }
 
+    /**
+     * @since 9.1
+     */
+    @Test
+    public void testWorkflowCleanUp() throws JsonProcessingException, IOException, InterruptedException {
+        DocumentModel note = RestServerInit.getNote(0, session);
+        final int max = 5;
+        for (int i = 0; i < max; i++) {
+            ClientResponse response = getResponse(RequestType.POST, "/workflow", getCreateAndStartWorkflowBodyContent(
+                    "ParallelDocumentReview", Collections.singletonList(note.getId())));
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            final String createdWorflowInstanceId = node.get("id").getTextValue();
+            // Cancel the workflow
+            response = getResponse(RequestType.DELETE, "/workflow/" + createdWorflowInstanceId);
+            assertEquals(Response.Status.NO_CONTENT.getStatusCode(), response.getStatus());
+        }
+        DocumentModelList cancelled = session.query(
+                "SELECT ecm:uuid FROM DocumentRoute WHERE ecm:currentLifeCycleState = 'canceled'");
+        assertEquals(max, cancelled.size());
+
+        EventContext eventContext = new EventContextImpl();
+        eventContext.setProperty("category", "workflowInstancesCleanup");
+        Event event = new EventImpl(DocumentRoutingWorkflowInstancesCleanup.CLEANUP_WORKFLOW_EVENT_NAME, eventContext);
+        EventService eventService = Framework.getService(EventService.class);
+        eventService.fireEvent(event);
+
+        awaitCleanupWorks();
+
+        cancelled = session.query(
+                "SELECT ecm:uuid FROM DocumentRoute WHERE ecm:currentLifeCycleState = 'canceled'");
+        assertTrue(cancelled.isEmpty());
+
+    }
+
+    /**
+     * @since 9.1
+     */
+    protected void awaitCleanupWorks() throws InterruptedException {
+        TransactionHelper.commitOrRollbackTransaction();
+        TransactionHelper.startTransaction();
+
+        workManager.awaitCompletion("default", 10000, TimeUnit.MILLISECONDS);
```

Comment:
`txFeature.nextTransaction()` is 10 minutes =>  need to use `nextTransaction(long duration, TimeUnit unit)`
Moreover the `awaitCompletion` method which takes the queue id as argument seems to do more than `txFeature.nextTransaction()`  => so, not sure replacing `workManager.awaitCompletion` by `txFeature.nextTransaction` does exactly the same thing